src/patchelf.cc

```diff
@@ -1885,9 +1906,186 @@ void ElfFile<ElfFileParamNames>::clearSymbolVersions(const std::set<std::string>
         }
     }
     changed = true;
-    this->rewriteSections();
+    rewriteSections();
 }
 
+struct symstr {
+    const char *str;
+    size_t len;
+};
+
+// Elf32_Off, Elf32_Word, Elf54_Word = unsigned int
+// Elf64_Off, Elf64_Xword = unsigned long (or long long on 32 bit)
+//typedef variant<Elf32_Off*,Elf32_Word*,Elf64_Off*,Elf64_Word*,Elf64_Xword*> symref;
+//typedef variant<unsigned int*, unsigned long*> symref;
+typedef variant<uint16_t*, uint32_t*, uint64_t*> symref;
+
+struct Comparator {
+    using is_transparent = void;
+    bool operator()(const symstr& a, const symstr& b) const
+    {
+        return (a.str+a.len < b.str+b.len);
+    }
+
+    bool operator()(const char* a, const symstr& b) const
+    {
+        return (a < b.str);
+    }
+
+    bool operator()(const symstr& a, const char* b) const
+    {
+        return (a.str+a.len < b);
+    }
+};
+
+#include <argz.h>
+
+class mymap : public map<symstr, forward_list<symref>, Comparator> {
```

Comment:
Perhaps one would want a better name for this class

Reply:
What would you suggest? symmap or such?